### PR TITLE
POC refactor of VideoGallery to reduce props

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What

Remove props related to videoStream and screenShareStream from VideoGallery and offload rendering code of remote video/ screenshare and local video/ screenshare to a callback prop.

# Why

Remove smart behaviour from VideoGallery

# How Tested

Sample and storybook

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Comment by mgamis-msft** [comment]
I was able to go further with the refactor to remove logic for dominantSpeakers and which participants goes to the grid/horizontal gallery such that the VideoGallery api looks like
```
    <VideoGallery
      localVideoTile={gridTiles.length > 0 ? localVideoTile : undefined}
      gridTiles={gridTiles.length > 0 ? gridTiles : [localVideoTile]}
      horizontalGalleryTiles={horizontalGalleryParticipantTiles}
      styles={VideoGalleryStyles}
    />
```
But I'm not sure if we want to go that far. Please let me know and I can update this PR.

File: packages/react-components/src/components/VideoGallery.tsx [hunk 1]
```diff
@@ -78,29 +66,14 @@ export interface VideoGalleryProps {
...
-
-  /**
-   * Whether to display a mute icon beside the user's display name.
-   * @defaultValue `true`
-   */
-  showMuteIndicator?: boolean;
+  /** Callback to render participants in VideoGallery */
+  onRenderTile: (
+    props: VideoGalleryLocalParticipant | VideoGalleryRemoteParticipant,
+    type: 'participant' | 'screenshare' | 'localParticipant' | 'localScreenshare'
```
Comment (alcail) [inline packages/react-components/src/components/VideoGallery.tsx:72]: just curious if we should have remote/local or just, as you did, no-prefix/local

File: packages/react-composites/src/composites/CallComposite/components/MediaGallery.tsx [hunk 2]
```diff
@@ -48,29 +48,59 @@ export const MediaGallery = (props: MediaGalleryProps): JSX.Element => {
...
     }
     setIsButtonStatusSynced(true);
   }, [isButtonStatusSynced, isPreviewCameraOn, props]);
 
+  const onRenderTile = useCallback(
+    (
+      participant: VideoGalleryLocalParticipant | VideoGalleryRemoteParticipant,
+      type: 'participant' | 'screenshare' | 'localParticipant' | 'localScreenshare'
+    ): JSX.Element => {
+      if (type === 'screenshare') {
```
Comment (alcail) [inline packages/react-composites/src/composites/CallComposite/components/MediaGallery.tsx:73]: nit: switch loop might be better then if ... else if ...

**Review by alcail — APPROVED** [review]

File: packages/react-components/src/components/VideoGallery.tsx (hunk 1 above)
Comment (mgamis-msft) [reply packages/react-components/src/components/VideoGallery.tsx:72]: Looking into this...

File: packages/communication-react/review/communication-react.api.md [hunk 3]
```diff
@@ -2025,18 +2053,9 @@ export interface VideoGalleryProps {
...
     layout?: 'default' | 'floatingLocalVideo';
     localParticipant: VideoGalleryLocalParticipant;
-    localVideoViewOption?: VideoStreamOptions;
-    onCreateLocalStreamView?: (options?: VideoStreamOptions) => Promise<void>;
-    onCreateRemoteStreamView?: (userId: string, options?: VideoStreamOptions) => Promise<void>;
-    onDisposeLocalStreamView?: () => void;
-    // (undocumented)
-    onDisposeRemoteStreamView?: (userId: string) => Promise<void>;
-    onRenderAvatar?: OnRenderAvatarCallback;
-    onRenderLocalVideoTile?: (localParticipant: VideoGalleryLocalParticipant) => JSX.Element;
```
Comment (PorterNan) [inline packages/communication-react/review/communication-react.api.md:2035]: For handlers - we should not remove them from videoGallery props, it's still here and pass down to the VideoTile

**Comment by mgamis-msft** [comment]
Closing this PR and revisit post GA.

**Pull request closed without merging**